agent/job_runner.go

```diff
@@ -392,6 +394,33 @@ func (r *JobRunner) normalizeVerificationBehavior(behavior string) (string, erro
 	}
 }
 
+func filterEnv(env map[string]string, allowedVariables []string) []string {
+	var deniedVariables []string
+
+	// Filter out if the user has unlisted environment variables
+	for k := range env {
+		isDenied := true
+
+		for _, v := range allowedVariables { // Loop through allowed environment variables list
```

Comment:
Remove that comment. Feels redundant as the code is clean.
```suggestion
		for _, v := range allowedVariables {
```